src/infra/ChatAnalyticsReporter.ts

```diff
@@ -0,0 +1,74 @@
+import type { EventPayload } from '../models/chat';
+import type ChatAnalyticsConfig from '../models/config/ChatAnalyticsConfig';
+
+interface EventsAPIResponse {
+  id: string;
+}
+
+/**
+ * A class to report chat analytics data. Uses the provided API key, environment,
+ * and partition to determine the endpoint for reporting events.
+ */
+export default class ChatAnalyticsReporter {
+  public readonly apiKey: string;
+  public readonly env: NonNullable<ChatAnalyticsConfig['env']>;
+  public readonly partition: NonNullable<ChatAnalyticsConfig['partition']>;
+  public readonly endpoint: string;
+
+  private endpoints: Record<
+    NonNullable<ChatAnalyticsConfig['partition']>,
+    Record<NonNullable<ChatAnalyticsConfig['env']>, string>
+  > = {
+    US: {
+      PROD: 'https://www.us.yextevents.com/accounts/me/events',
+      SANDBOX: 'https://www.sbx.us.yextevents.com/accounts/me/events',
+      QA: 'https://www.qa.us.yextevents.com/accounts/me/events',
```

Comment:
I don't think we publish our QA/DEV urls generally

Reply:
In search we export this endpoints object: https://github.com/yext/search-core/blob/master/src/models/core/Endpoints.ts

Reply:
and we build the object here: https://github.com/yext/search-core/blob/master/src/provideEndpoints.ts#L13

Reply:
which lets you pass in other environments that are un-documented

Reply:
Ok I can get a rid of the QA URL. I can also add the ability to manually specify a `domain`. But I think we should keep the current default behavior of accepting an `env` and a `partition` and then figuring out the URL for you, because otherwise it's very annoying to figure out what domain you're actually looking for. Does that work?

Reply:
Also changed this to "region"